### PR TITLE
Ensure proper initialization of a ”Me” pointer in a Closure created for a block within constructor body.

### LINKED ISSUE #4899: Closure isn't initialized properly within a Try block inside a constructor
Ported from DevDiv 1207506

Compile and run the following code:
```
Module Module1

    Sub Main()
        Dim f As New FpB(100, 100)
        System.Console.WriteLine(f.FPixels.Length)
    End Sub

End Module

Public Class FpB
    Public Property FPixels() As FloatPointF(,)
        Get
            Return m_FPixels
        End Get
        Set(value As FloatPointF(,))
            m_FPixels = value
        End Set
    End Property
    Private m_FPixels As FloatPointF(,)


    Public Sub New(width As Integer, height As Integer)
        Try
            Dim w As Integer = width
            Dim h As Integer = height
            Me.FPixels = New FloatPointF(w - 1, h - 1) {}
            CallDelegate(Sub(y)
                             Dim x = Math.Min(0, w - 1)
                             System.Console.WriteLine(Me.FPixels(x, y).X)
                         End Sub)
        Catch ex As Exception
            System.Console.WriteLine(ex.Message)
        End Try
    End Sub

    Sub CallDelegate(d As Action(Of Integer))
        d(1)
    End Sub
End Class

Public Structure FloatPointF
    Public X As Single
    Public Y As Single
End Structure
```

A NullReferenceException is thrown inside FPixels getter.

Doesn't happen when Try block is extracted into a separate method, or when declaration of ```w``` is moved out of the Try block.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #4899.

@VSadov, @gafter, @jaredpar, @agocke Please review.

### CODE REVIEW

File: src/Compilers/VisualBasic/Test/Semantic/Semantics/LambdaTests.vb [hunk 1]
```diff
@@ -2097,5 +2097,70 @@ End Class
...
+            Return m_FPixels
+        End Get
+        Set(value As FloatPointF(,))
+            m_FPixels = value
+        End Set
+    End Property
+    Private m_FPixels As FloatPointF(,)
+
+
+    Public Sub New(width As Integer, height As Integer)
```
Comment (jaredpar) [inline src/Compilers/VisualBasic/Test/Semantic/Semantics/LambdaTests.vb]: Do we have any tests where we capture `Me` and chain to a base ctor which has non-trivial expressions in the `MyBase.New` call?  For example calling:

``` vb
Sub New(p As Object)
  MyBase.New(p.ToString())
  Dim l = Function() Me.ToString() 
```

I remember that being a source of problem for the native compiler.

**Comment by jaredpar** [comment]
:+1: 

Does C# need a similar change?

**Comment by AlekseyTs** [comment]
C# already has this logic.

File: src/Compilers/VisualBasic/Test/Semantic/Semantics/LambdaTests.vb (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Semantic/Semantics/LambdaTests.vb]: It is very likely we have something like this. The existing code ensures that we are not going to attempt to capture Me until constructor initializer is called, regardless of its arguments.

**Comment by VSadov** [comment]
LGTM

**Comment by AlekseyTs** [comment]
@dotnet-bot Please test this.

**Comment by AlekseyTs** [comment]
@dotnet-bot Test this please.

**Pull request merged**